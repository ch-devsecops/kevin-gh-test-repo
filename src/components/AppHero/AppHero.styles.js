import { ThemeStyles } from '@honda-canada/design-system-react';
import css from '@styled-system/css';
import variant from '@styled-system/variant';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('ContentWrapper')(
  css({
    display: 'flex',
    flexDirection: 'column',
    height: ['85%', '85%', '100%'],
    position: ['absolute', 'absolute', 'relative'],
    top: ['20%', '25%', 0],
    left: [0, 0, 'unset'],
    mt: [6, 8, 'unset'],
  }),
  variant({
    prop: 'verticalAlignment',
    variants: {
      top: {
        justifyContent: ['space-evenly', 'space-evenly', 'flex-start'],
      },
    },
  }),
);

themeStyles.addBaseStyles('Content')(
  ({ horizontalAlignment }) =>
    css({
      textAlign: ['center', 'center', horizontalAlignment],
      display: 'inline-block',
      position: ['relative', 'relative', 'absolute'],
    }),
  variant({
    prop: 'horizontalAlignment',
    variants: {
      left: {
        left: 0,
      },
      right: {
        right: 0,
      },
      center: {
        margin: '0 auto',
        left: 0,
        right: 0,
      },
    },
  }),
  variant({
    prop: 'verticalAlignment',
    variants: {
      top: {
        top: 0,
      },
      bottom: {
        bottom: 0,
      },
    },
  }),
);

themeStyles.addBaseStyles('SubTitle')(
  css({
    maxWidth: ['370px', '400px', '500px'],
    lineHeight: ['22px', '22px', '26px'],
    margin: '8px auto',
    color: 'white',
    display: 'inline-block',
  }),
);

themeStyles.addBaseStyles('ButtonsWrapper')(({ horizontalAlignment }) => {
  let justifyContent = 'center';

  if (horizontalAlignment === 'left') {
    justifyContent = 'flex-start';
  }

  if (horizontalAlignment === 'right') {
    justifyContent = 'flex-end';
  }
  return css({
    display: 'flex',
    flexWrap: ['wrap', 'wrap', 'nowrap'],
    alignItems: 'center',
    maxWidth: ['100vw', '500px', 'unset'],
    justifyContent: ['center', 'center', justifyContent],
    margin: ['2px auto', '2px auto', '4px auto'],
    // workaround to keep responsive visibility div from
    // wrecking justify-content
    '> div.fresnel-container': {
      display: ['none', 'none', 'inline-block'],
    },
  });
});

themeStyles.addBaseStyles('BackgroundContainer')(({ mobileImageSource, desktopImageSource }) =>
  css({
    position: 'absolute',
    left: '0',
    top: '0',
    width: '100%',
    height: '100%',
    background: [
      `url('${mobileImageSource}') center top / cover no-repeat`,
      `url('${mobileImageSource}') center top / cover no-repeat`,
      `url('${desktopImageSource}') center center / cover no-repeat`,
    ],
  }),
);

themeStyles.addBaseStyles('ImageLink')(
  css({
    marginTop: [2, 4, undefined],
    width: ['40%', '40%', 'unset'],
    '&:hover': {
      borderColor: 'transparent',
    },
    '> img': {
      maxWidth: '100%',
    },
  }),
);

export default themeStyles;
