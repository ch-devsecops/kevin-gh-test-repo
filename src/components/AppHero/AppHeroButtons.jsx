import { Button, Link } from '@honda-canada/design-system-react';
import { Image, isEditorActive } from '@sitecore-jss/sitecore-jss-react';

import PropTypes from 'prop-types';
import React from 'react';
import RoutableSitecoreLink from '../RoutableSitecoreLink';
import themeStyles from './AppHero.styles';
import { getGtmTagValue } from '../../utils/gtmEvents';

const ImageLink = themeStyles.apply(Link, 'ImageLink');

const AppHeroButtons = ({ buttons, gtmTitle, gtmInteractionType, gtmComponentType }) => {
  if (!buttons) return null;
  const gtmTags = {
    'data-gtm-title': getGtmTagValue(!isEditorActive() && gtmTitle),
    'data-gtm-interaction-type': getGtmTagValue(gtmInteractionType),
    'data-gtm-component-type': getGtmTagValue(gtmComponentType),
  };
  return (
    <>
      {buttons.map((button, index) => {
        if (button.JSSImageField) {
          return (
            <ImageLink
              as={RoutableSitecoreLink}
              aria-label={button.ariaLabel}
              field={button.JSSLinkField}
              key={index.toString()}
              mr={[4, 6, index === buttons.length - 1 ? 0 : 3]}
              height={`${button.JSSImageField.value.height}px`}
              gtmTags={gtmTags}
              {...button.props}
            >
              <Image field={button.JSSImageField} style={{ float: index ? 'left' : 'right' }} />
            </ImageLink>
          );
        }
        return (
          <Button
            as={RoutableSitecoreLink}
            aria-label={button.ariaLabel}
            field={button.JSSLinkField}
            styling={`${button.styling}Dark`}
            key={index.toString()}
            mt={[2, 4, undefined]}
            gtmTags={gtmTags}
            {...button.props}
          />
        );
      })}
    </>
  );
};

AppHeroButtons.propTypes = {
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      JSSImage: PropTypes.shape({
        src: PropTypes.string,
        alt: PropTypes.string,
        width: PropTypes.string,
        height: PropTypes.string,
      }),
      JSSField: PropTypes.shape({
        value: PropTypes.string,
      }),
      text: PropTypes.string,
      styling: PropTypes.string,
    }),
  ),
  gtmTitle: PropTypes.string,
  gtmComponentType: PropTypes.string,
  gtmInteractionType: PropTypes.string,
};

export default AppHeroButtons;
