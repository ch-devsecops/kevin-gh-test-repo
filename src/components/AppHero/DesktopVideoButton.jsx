import React from 'react';
import PropTypes from 'prop-types';
import { Box, heroStyles, Icon, Media } from '@honda-canada/design-system-react';

const VideoButton = heroStyles.apply('button', 'VideoButton');

const getPositionLeft = (h, v) => h === 'right' && v === 'bottom' && 0;

const getPositionRight = (h, v) => {
  if (v === 'top' || (v === 'bottom' && h !== 'right')) return 0;
  if (v === 'bottom' && h !== 'right') return 0;

  return undefined;
};

const DesktopVideoButton = ({ video, horizontalAlignment, verticalAlignment }) =>
  video ? (
    <Media greaterThan="smallDesktop">
      <Box
        position="absolute"
        bottom={0}
        zIndex={1}
        px="s"
        left={getPositionLeft(horizontalAlignment, verticalAlignment)}
        right={getPositionRight(horizontalAlignment, verticalAlignment)}
      >
        <VideoButton onClick={video.onPlay} aria-label={video.ariaLabel} className="videoButton">
          <Icon name="play" iconSize="large" background="white" />
        </VideoButton>
      </Box>
    </Media>
  ) : null;

DesktopVideoButton.propTypes = {
  video: PropTypes.shape({
    ariaLabel: PropTypes.string,
    onPlay: PropTypes.func,
  }),
  horizontalAlignment: PropTypes.oneOf(['left', 'center', 'right']),
  verticalAlignment: PropTypes.oneOf(['top', 'bottom']),
};

export default DesktopVideoButton;
