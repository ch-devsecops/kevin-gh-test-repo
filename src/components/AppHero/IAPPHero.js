import PropTypes from 'prop-types';

export default {
  seoH1: PropTypes.shape({
    field: PropTypes.shape({ value: PropTypes.string }),
  }).isRequired,
  title: PropTypes.shape({
    field: PropTypes.shape({ value: PropTypes.string }),
  }).isRequired,
  subtitle: PropTypes.shape({
    field: PropTypes.shape({ value: PropTypes.string }),
  }),
  desktopImage: PropTypes.shape({
    field: PropTypes.shape({
      value: {
        src: PropTypes.string,
      },
    }),
  }),
  mobileImage: PropTypes.shape({
    field: PropTypes.shape({
      value: {
        src: PropTypes.string,
      },
    }),
  }),
  ctaImage1: PropTypes.shape({
    field: PropTypes.shape({
      value: {
        src: PropTypes.string,
      },
    }),
  }),
  ctaImage2: PropTypes.shape({
    field: PropTypes.shape({
      value: {
        src: PropTypes.string,
      },
    }),
  }),
  ctaImageLink1: PropTypes.shape({
    field: PropTypes.shape({
      value: {
        href: PropTypes.string,
      },
    }),
  }),
  ctaImageLink2: PropTypes.shape({
    field: PropTypes.shape({
      value: {
        href: PropTypes.string,
      },
    }),
  }),
  ctaUrl: PropTypes.shape({
    field: PropTypes.shape({
      value: {
        href: PropTypes.string,
      },
    }),
  }),
  ctaType: PropTypes.shape({
    field: PropTypes.shape({
      value: PropTypes.string,
    }),
  }),
  badgeImage: PropTypes.shape({
    field: PropTypes.shape({
      value: {
        src: PropTypes.string,
      },
    }),
  }),
  videoUrl: PropTypes.shape({
    value: PropTypes.shape({
      anchor: PropTypes.string,
      href: PropTypes.string,
      linktype: PropTypes.string,
      text: PropTypes.string,
      url: PropTypes.string,
      target: PropTypes.string,
    }),
  }),
  gtmTitle: PropTypes.shape({
    value: PropTypes.string,
  }),
  gtmCategory: PropTypes.shape({
    value: {
      src: PropTypes.string,
    },
  }),
  gtmInteractionType: PropTypes.shape({
    value: {
      src: PropTypes.string,
    },
  }),
};
