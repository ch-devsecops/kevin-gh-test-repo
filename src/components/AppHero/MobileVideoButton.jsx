import { Box, Icon, Media, heroStyles } from '@honda-canada/design-system-react';

import PropTypes from 'prop-types';
import React from 'react';

const VideoButton = heroStyles.apply('button', 'VideoButton');

const MobileVideoButton = ({ video }) =>
  video ? (
    <Media lessThan="desktop">
      <Box pt={['xl', 'big']} display="flex" justifyContent="center">
        <VideoButton onClick={video.onPlay} aria-label={video.ariaLabel} className="videoButton">
          <Icon name="play" iconSize="large" background="white" />
        </VideoButton>
      </Box>
    </Media>
  ) : null;

MobileVideoButton.propTypes = {
  video: PropTypes.shape({
    ariaLabel: PropTypes.string,
    onPlay: PropTypes.func,
  }),
};

export default MobileVideoButton;
