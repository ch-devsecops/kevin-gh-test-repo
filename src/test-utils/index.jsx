import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
