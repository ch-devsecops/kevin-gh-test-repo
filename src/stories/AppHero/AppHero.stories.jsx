import React from 'react';
import AppHero from '../../components/AppHero';
import { mockData as storyProps } from './componentProps';

export default {
  title: 'AppHero',
  component: AppHero,
};

const Template = args => <AppHero {...args} />;

export const Default = Template.bind({});

Default.args = storyProps;
