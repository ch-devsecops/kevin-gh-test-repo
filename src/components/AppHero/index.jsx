import React from 'react';
import {
  Box,
  Column,
  Copy,
  heroStyles,
  Optional,
  Row,
  useOptionalVideo,
  Wrapper,
} from '@honda-canada/design-system-react';
import isValidUrlOrRelativeUrl from '@honda-canada/js-utilities/lib/isValidUrlOrRelativeUrl';
import { compiler } from 'markdown-to-jsx';
import { getHorizontalAlignment, getVerticalAlignment } from '../Hero';
import IAPPHero from './IAPPHero';
import { getTitleComponent, stripMarkdownHeading } from '../../utils/markdown';
import { getVideoProps, mapGTMCategory } from '../../utils/sitecoreFields';

import AppHeroButtons from './AppHeroButtons';
import DesktopVideoButton from './DesktopVideoButton';
import HeroBadge from '../HeroBadge';
import MobileVideoButton from './MobileVideoButton';
import themeStyles from './AppHero.styles';
import { wrapJSSFields } from '../../utils/wrapJSSFields';
import { getGtmTagValue } from '../../utils/gtmEvents';

const Container = heroStyles.apply(Box, 'Container');
const BackgroundContainer = themeStyles.apply('span', 'BackgroundContainer');
const HeroGradient = heroStyles.apply(Box, 'Gradient');

const ContentWrapper = themeStyles.apply(Row, 'ContentWrapper');
const Content = themeStyles.apply(Column, 'Content');
const SubTitle = themeStyles.apply(Copy, 'SubTitle');
const ButtonsWrapper = themeStyles.apply(Box, 'ButtonsWrapper');

const AppHero = ({ fields, params, rendering }) => {
  if (!fields || !params) return null;

  const {
    seoH1,
    title,
    subtitle,
    desktopImage,
    mobileImage,
    ctaImage1,
    ctaImage2,
    ctaImageLink1,
    ctaImageLink2,
    ctaUrl,
    ctaType,
    badgeImage,
    videoUrl,
    gtmTitle,
    gtmCategory,
    gtmInteractionType,
  } = wrapJSSFields(fields);

  const video = getVideoProps(
    videoUrl?.value?.href,
    videoUrl?.value?.title,
    'close video modal',
    videoUrl?.value?.text,
  );

  // eslint-disable-next-line react-hooks/rules-of-hooks
  const { optionalVideo } = useOptionalVideo(video);
  const horizontalAlignment = getHorizontalAlignment(params?.contentPosition);
  const verticalAlignment = getVerticalAlignment(params?.contentPosition);
  const isShort = params?.heroStyleType === 'Short';
  const desktopImageSource = isValidUrlOrRelativeUrl(desktopImage?.getProp('src')) ? desktopImage?.getProp('src') : '';
  const mobileImageSource = isValidUrlOrRelativeUrl(mobileImage?.getProp('src')) ? mobileImage?.getProp('src') : '';
  const imageAltText = desktopImage?.getProp('alt');
  const gtmTags = {
    category: mapGTMCategory(gtmCategory),
    type: rendering?.componentName,
  };
  const TitleComponent = getTitleComponent(title?.value);
  const buttonsArray = [];

  if (ctaImage1?.getProp('src')) {
    buttonsArray.push({
      JSSLinkField: ctaImageLink1.field,
      JSSImageField: ctaImage1.field,
    });
  }

  if (ctaImage2?.getProp('src')) {
    buttonsArray.push({
      JSSLinkField: ctaImageLink2.field,
      JSSImageField: ctaImage2.field,
    });
  }

  if (ctaUrl?.getProp('url')) {
    buttonsArray.push({
      JSSLinkField: ctaUrl.field,
      styling: ctaType.value?.toLowerCase() || 'secondary',
      text: ctaUrl.value?.text,
      ariaLabel: ctaUrl.value?.title,
    });
  }

  // eslint-disable-next-line react/no-unstable-nested-components
  const Buttons = () => (
    <AppHeroButtons
      buttons={buttonsArray}
      gtmTitle={gtmTitle?.value}
      gtmInteractionType={gtmInteractionType?.value}
      gtmComponentType={rendering?.componentName}
    />
  );

  return (
    <Container
      isShort={isShort}
      data-gtm-category={getGtmTagValue(gtmTags?.category)}
      data-gtm-component-type={getGtmTagValue(gtmTags?.type)}
      verticalAlignment={verticalAlignment}
    >
      <BackgroundContainer
        isShort={isShort}
        desktopImageSource={desktopImageSource}
        mobileImageSource={mobileImageSource}
        aria-label={imageAltText}
        alt={imageAltText}
        role="img"
      />
      <HeroGradient verticalAlignment={verticalAlignment} />
      <Wrapper height="100%" position="relative">
        {optionalVideo?.src && (
          <DesktopVideoButton
            video={video}
            horizontalAlignment={horizontalAlignment}
            verticalAlignment={verticalAlignment}
          />
        )}
        {optionalVideo?.src && <MobileVideoButton video={video} />}
        <ContentWrapper horizontalAlignment={horizontalAlignment} verticalAlignment={verticalAlignment}>
          <Content
            width={[1, 1, 7 / 12]}
            horizontalAlignment={horizontalAlignment}
            verticalAlignment={verticalAlignment}
          >
            <HeroBadge image={badgeImage} horizontalAlignment={horizontalAlignment} />
            <TitleComponent
              as={seoH1?.value === 'Title' ? 'h1' : 'p'}
              color="white"
              mt={['8px', '8px', '16px']}
              width="100%"
            >
              {compiler(stripMarkdownHeading(title?.value))}
            </TitleComponent>
            <Optional when={subtitle?.value}>
              <SubTitle as={seoH1?.value === 'Subtitle' && 'h1'}>
                {compiler(stripMarkdownHeading(subtitle?.value))}
              </SubTitle>
            </Optional>
            <ButtonsWrapper horizontalAlignment={horizontalAlignment}>
              <Buttons />
            </ButtonsWrapper>
          </Content>
        </ContentWrapper>
      </Wrapper>
    </Container>
  );
};

AppHero.propTypes = IAPPHero;
export default AppHero;
