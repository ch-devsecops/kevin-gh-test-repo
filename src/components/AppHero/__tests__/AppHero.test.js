import React from 'react';
import { axe, cleanup, jssRender } from 'test-utils';
import AppHero from '..';

afterEach(cleanup);

const mockData = require('../../../../sample-mock-data/routes/app-hero/en.json');
const emptyMockData = require('../../../../sample-mock-data/routes/app-hero/empty_en.json');
const mockEmptyValues = require('../../../../sample-mock-data/routes/app-hero/empty_values_en.json');

describe('<AppHero>', () => {
  test('should render correctly', () => {
    const { container } = jssRender(<AppHero {...mockData} />);

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeTruthy();
  });

  test('should return null if there is no fields', () => {
    const { container } = jssRender(<AppHero fields={null} />);

    expect(container).toMatchSnapshot();
  });
  test('should return null if the data values are empty', () => {
    const { container } = jssRender(<AppHero {...mockEmptyValues} />);

    expect(container).toMatchSnapshot();
  });

  test('should return null if the data is empty', () => {
    const { container } = jssRender(<AppHero {...emptyMockData} />);
    expect(container).toMatchSnapshot();
  });

  test('should have no accessibility violations', async () => {
    const { container } = jssRender(<AppHero {...mockData} />);

    expect(container).toMatchSnapshot();
    expect(await axe(container)).toHaveNoViolations();
  });
});
