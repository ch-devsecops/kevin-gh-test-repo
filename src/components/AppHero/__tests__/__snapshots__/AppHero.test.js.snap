// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHero> should have no accessibility violations 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 fOWcBp"
    data-gtm-category="Others"
  >
    <span
      alt="Honda GX engine 3 year warranty badge"
      aria-label="Honda GX engine 3 year warranty badge"
      class="sc-1k44dfv-0 cdiPwK"
      role="img"
    />
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 fgzTle"
    />
    <div
      class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 sOsmN"
      height="100%"
    >
      <div
        class="fresnel-container fresnel-greaterThan-smallDesktop "
      >
        <div
          class="Box-sc-1bsf784-0 fpfBkZ"
        >
          <button
            aria-label="video alt text"
            class="sc-1k44dfv-0 ihyQFC videoButton"
          />
        </div>
      </div>
      <div
        class="fresnel-container fresnel-lessThan-desktop "
      >
        <div
          class="Box-sc-1bsf784-0 coYchu"
          display="flex"
        >
          <button
            aria-label="video alt text"
            class="sc-1k44dfv-0 ihyQFC videoButton"
          />
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 jGVkdm sc-1k44dfv-0 dvPCJu"
        display="flex"
        width="100%"
      >
        <div
          class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 jspGaB"
          width="1,1,0.5833333333333334"
        >
          <p
            class="sc-1k44dfv-0 jMpDHg"
            color="white"
            width="100%"
          >
            <span>
              Engine Link
            </span>
          </p>
          <p
            class="sc-1k44dfv-0 sc-1k44dfv-0 eDqEXM"
            color="typographyDefault"
            font-family="default"
            font-weight="regular"
          >
            <span>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna
            </span>
          </p>
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 bZXcvK"
          >
            <a
              class="Link-sc-16wd4a0-0 sc-1k44dfv-0 kHgxtN"
              href="https://www.google.com"
              target=""
              title="test cta url 1"
            >
              <img
                alt="AppStore"
                height="43"
                src="https://engine-dit-01-dmm.engine.honda.ca/-/media/Images/app-store-badge.png?h=43&iar=0&w=132&rev=1578264cb998483fb99082fcf11e70ed&hash=210B7F73808A32B461CC5DDB2FFB3EA7"
                style="float: right;"
                width="132"
              />
            </a>
            <a
              class="Link-sc-16wd4a0-0 sc-1k44dfv-0 kXSNwc"
              href="https://www.google.com"
              target=""
              title="Test cta url 2 "
            >
              <img
                alt="google"
                height="45"
                src="https://engine-dit-01-dmm.engine.honda.ca/-/media/Images/google-play-badge.png?h=45&iar=0&w=152&rev=c81d22af8430455a981fc8fcb24a3053&hash=8C53BBABCC8BCD212A2035DA488633D4"
                style="float: left;"
                width="152"
              />
            </a>
            <a
              class="Button-pulw6t-0 eSCMqV"
              data-testid="dsr-button-component"
              href="https://engine-dit-01-dmm.engine.honda.ca/"
              target=""
              title="test enroll"
            >
              Enroll
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AppHero> should render correctly 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 fOWcBp"
    data-gtm-category="Others"
  >
    <span
      alt="Honda GX engine 3 year warranty badge"
      aria-label="Honda GX engine 3 year warranty badge"
      class="sc-1k44dfv-0 cdiPwK"
      role="img"
    />
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 fgzTle"
    />
    <div
      class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 sOsmN"
      height="100%"
    >
      <div
        class="fresnel-container fresnel-greaterThan-smallDesktop "
      >
        <div
          class="Box-sc-1bsf784-0 fpfBkZ"
        >
          <button
            aria-label="video alt text"
            class="sc-1k44dfv-0 ihyQFC videoButton"
          />
        </div>
      </div>
      <div
        class="fresnel-container fresnel-lessThan-desktop "
      >
        <div
          class="Box-sc-1bsf784-0 coYchu"
          display="flex"
        >
          <button
            aria-label="video alt text"
            class="sc-1k44dfv-0 ihyQFC videoButton"
          />
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 jGVkdm sc-1k44dfv-0 dvPCJu"
        display="flex"
        width="100%"
      >
        <div
          class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 jspGaB"
          width="1,1,0.5833333333333334"
        >
          <p
            class="sc-1k44dfv-0 jMpDHg"
            color="white"
            width="100%"
          >
            <span>
              Engine Link
            </span>
          </p>
          <p
            class="sc-1k44dfv-0 sc-1k44dfv-0 eDqEXM"
            color="typographyDefault"
            font-family="default"
            font-weight="regular"
          >
            <span>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna
            </span>
          </p>
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 bZXcvK"
          >
            <a
              class="Link-sc-16wd4a0-0 sc-1k44dfv-0 kHgxtN"
              href="https://www.google.com"
              target=""
              title="test cta url 1"
            >
              <img
                alt="AppStore"
                height="43"
                src="https://engine-dit-01-dmm.engine.honda.ca/-/media/Images/app-store-badge.png?h=43&iar=0&w=132&rev=1578264cb998483fb99082fcf11e70ed&hash=210B7F73808A32B461CC5DDB2FFB3EA7"
                style="float: right;"
                width="132"
              />
            </a>
            <a
              class="Link-sc-16wd4a0-0 sc-1k44dfv-0 kXSNwc"
              href="https://www.google.com"
              target=""
              title="Test cta url 2 "
            >
              <img
                alt="google"
                height="45"
                src="https://engine-dit-01-dmm.engine.honda.ca/-/media/Images/google-play-badge.png?h=45&iar=0&w=152&rev=c81d22af8430455a981fc8fcb24a3053&hash=8C53BBABCC8BCD212A2035DA488633D4"
                style="float: left;"
                width="152"
              />
            </a>
            <a
              class="Button-pulw6t-0 eSCMqV"
              data-testid="dsr-button-component"
              href="https://engine-dit-01-dmm.engine.honda.ca/"
              target=""
              title="test enroll"
            >
              Enroll
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AppHero> should return null if the data is empty 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 fOWcBp"
    data-gtm-category="Others"
  >
    <span
      class="sc-1k44dfv-0 cCXBR"
      role="img"
    />
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 gqxPaN"
    />
    <div
      class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 sOsmN"
      height="100%"
    >
      <div
        class="Box-sc-1bsf784-0 jGVkdm sc-1k44dfv-0 hwakGZ"
        display="flex"
        width="100%"
      >
        <div
          class="Box-sc-1bsf784-0 Column-t2p57o-0 sc-1k44dfv-0 iMrgcb"
          width="1,1,0.5833333333333334"
        >
          <p
            class="sc-1k44dfv-0 jMpDHg"
            color="white"
            width="100%"
          >
            <span />
          </p>
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 dpEQWN"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AppHero> should return null if the data values are empty 1`] = `<div />`;

exports[`<AppHero> should return null if there is no fields 1`] = `<div />`;
